src/main/java/se/liu/ida/hefquin/engine/queryplan/utils/TextBasedLogicalPlanPrinterImpl2.java

```diff
@@ -0,0 +1,63 @@
+package se.liu.ida.hefquin.engine.queryplan.utils;
+
+import java.io.PrintStream;
+
+import se.liu.ida.hefquin.engine.queryplan.logical.LogicalPlan;
+import se.liu.ida.hefquin.engine.queryplan.logical.LogicalPlanVisitor;
+import se.liu.ida.hefquin.engine.queryplan.logical.LogicalPlanWalker;
+import se.liu.ida.hefquin.engine.queryplan.logical.impl.*;
+import se.liu.ida.hefquin.engine.utils.IndentingPrintStream;
+
+public class TextBasedLogicalPlanPrinterImpl2 implements LogicalPlanPrinter
+{
+	// The string represents '|  '.
+	private static String levelIndentBase = "\u2502  ";
+	// The string represents '├──'.
+	private static String nonLastChildIndentBase = "\u251C\u2500\u2500";
+	// The string represents '└──'.
+	private static String lastChildIndentBase = "\u2514\u2500\u2500";
+		
+	@Override
+	public void print( final LogicalPlan plan, final PrintStream out ) {
+		final IndentingPrintStream iOut = new IndentingPrintStream(out);
```

Comment:
Is it still necessary to wrap the given `PrintStream` into an `IndentingPrintStream`? I don't think you are using the functionality of `IndentingPrintStream` anywhere here, are you?